### ISSUE TITLE
OSSL_STORE_attach() fails when ui_method is NULL

### ISSUE THREAD

**Issue by mouse07410** [openssl/openssl#12830]
MacOS 10.15.6, Xcode-11.7, current OpenSSL master.

It looks like `OSSL_STORE_attach()` in `crypto/store/store_lib.c` forces a call to `ossl_pw_set_ui_method()` here https://github.com/openssl/openssl/blob/b434b2c08d2025936fb8b7ece3a5908613333f6b/crypto/store/store_lib.c#L971, even if `ui_method` and/or `ui_data` are NULL. Which in turn fails assertion here: https://github.com/openssl/openssl/blob/b434b2c08d2025936fb8b7ece3a5908613333f6b/crypto/passphrase.c#L84

Here's the call to `OSSL_STORE_attach()`:
```c
. . . . .
     pkey = NULL;
     OSSL_STORE_CTX *cts;
     T(cts = OSSL_STORE_attach(bp, "file", NULL, NULL, NULL, NULL, NULL, NULL));
. . . . .
```

The error message is
```
	OSSL_STORE_attach:	crypto/passphrase.c:84: OpenSSL internal error: Assertion failed: data != NULL && ui_method != NULL
```

I stumbled upon this problem running GOST engine test: https://github.com/gost-engine/engine/blob/master/test_sign.c (see issue https://github.com/gost-engine/engine/issues/276)

@levitte 
I'm working on a small reproducer (which, hopefully, won't be a big deal), but in the meanwhile, perhaps, you could see what the problem is, and maybe even how to fix it.

**Comment by levitte** [comment]
Fix in #12831

**Comment by mouse07410** [comment]
I confirm that #12831 fixed the observed problem. @levitte thank you!
